docs/src/managing-packages.md

```diff
@@ -0,0 +1,298 @@
+  Updating `~/.julia/environments/v1.0/Project.toml`
+  [7876af07] + Example v0.5.1+ [`~/.julia/dev/Example`]
+  Updating `~/.julia/environments/v1.0/Manifest.toml`
+  [7876af07] + Example v0.5.1+ [`~/.julia/dev/Example`]
+```
+
+The `dev` command fetches a full clone of the package to `~/.julia/dev/` (the path can be changed by setting the environment variable `JULIA_PKG_DEVDIR`).
+When importing `Example` julia will now import it from `~/.julia/dev/Example` and whatever local changes have been made to the files in that path are consequently
+reflected in the code loaded. When we used `add` we said that we tracked the package repository, we here say that we track the path itself.
+Note that the package manager will never touch any of the files at a tracked path. It is therefore up to you to pull updates, change branches etc.
+If we try to `dev` a package at some branch that already exists at `~/.julia/dev/` the package manager we will simply use the existing path.
+For example:
+
+```
+(v1.0) pkg> dev Example
+  Updating git-repo `https://github.com/JuliaLang/Example.jl.git`
+[ Info: Path `/Users/kristoffer/.julia/dev/Example` exists and looks like the correct package, using existing path instead of cloning
+```
+
+Note the info message saying that it is using the existing path. As a general rule, the package manager will
+never touch files that are tracking a path.
+
+If `dev` is used on a local path, that path to that package is recorded and used when loading that package.
+The path will be recorded relative to the project file, unless it is given as an absolute path.
+
+To stop tracking a path and use the registered version again, use `free`
+
+```
+(v1.0) pkg> free Example
+ Resolving package versions...
+  Updating `~/.julia/environments/v1.0/Project.toml`
+  [7876af07] ↓ Example v0.5.1+ [`~/.julia/dev/Example`] ⇒ v0.5.1
+  Updating `~/.julia/environments/v1.0/Manifest.toml`
+  [7876af07] ↓ Example v0.5.1+ [`~/.julia/dev/Example`] ⇒ v0.5.1
+```
+
+It should be pointed out that by using `dev` your project is now inherently stateful.
+Its state depends on the current content of the files at the path and the manifest cannot be "instantiated" by someone else without
+knowing the exact content of all the packages that are tracking a path.
+
+Note that if you add a dependency to a package that tracks a local path, the Manifest (which contains the whole dependency graph) will become
+out of sync with the actual dependency graph. This means that the package will not be able to load that dependency since it is not recorded
+in the Manifest. To update sync the Manifest, use the REPL command `resolve`.
+
+## Removing packages
+
+Packages can be removed from the current project by using `pkg> rm Package`.
+This will only remove packages that exist in the project, to remove a package that only
+exists as a dependency use `pkg> rm --manifest DepPackage`.
+Note that this will remove all packages that depends on `DepPackage`.
```

Comment:
Do `resolve` and/or `gc` automatically remove dependencies that are no longer needed?